src/execution_plan/ops/op_project.c

```diff
@@ -22,6 +22,25 @@ static AR_ExpNode** _getOrderExpressions(OpBase *op) {
     return _getOrderExpressions(op->parent);
 }
 
+ /**
+  * @brief Counts the actual aliases inside the aliases array. 
+  * The aliases array is an array with size of the original projected values.
+  * It holds the mapping between a variable and its alias, or null if such alias does not exists.
+  * We need to use this function in order to allow a projected record to hold both its 
+  * projected variables values, and their alises values. If a record is allocated only with the
+  * length of the size of the expression variables count, it cannot store its alises values,
+  * and will cause a memory leak.
+  * @param  aliases: aliases mapping array
+  * @retval the amount of alised variables.
+  */
+int _actualAliasesCount(char** aliases){
+    int count = 0;
+    for (int i = 0; i < array_len(aliases); i++){
```

Comment:
It's better to avoid computing array_len with every iteration,
int alias_count = array_len(aliases);
for(int i = 0; i < alias_count; i++)...